### PR TITLE
Update for doxygen inline docs

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1751

### CODE REVIEW

File: sdk/core/azure-core/test/ut/md5.cpp [hunk 1]
```diff
@@ -118,3 +118,33 @@ TEST(Md5Hash, CtorDtor)
...
+  // Each created thread will wait from 0 to 3 milliseconds to start to make threads overlap
+  for (int counter = 0; counter < size; counter++)
+  {
+    pool.emplace_back(std::thread(hashThreadRoutine, counter % 4));
+  }
+  for (int counter = 0; counter < size; counter++)
+  {
+    pool[counter].join();
+  }
+}
```
Comment (Jinming-Hu) [inline sdk/core/azure-core/test/ut/md5.cpp:150]: Whoa, do it in another PR.

Comment (vhvb1989) [reply sdk/core/azure-core/test/ut/md5.cpp:150]: My bad

Comment (vhvb1989) [reply sdk/core/azure-core/test/ut/md5.cpp]: btw, thank you for the fix @Jinming-Hu , I saw you provided in the issue, I just took it and applied

File: cmake-modules/AzureDoxygen.cmake [hunk 2]
```diff
@@ -18,7 +18,7 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_LAYOUT_FILE ${CMAKE_SOURCE_DIR}/eng/docs/api/assets/DoxygenLayout.xml)
         set(DOXYGEN_RECURSIVE YES)
         set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ./README.md)
-        set(DOXYGEN_INLINE_SOURCES YES)
+        set(DOXYGEN_INLINE_SOURCES NO)
```
Comment (Jinming-Hu) [inline cmake-modules/AzureDoxygen.cmake:21]: Do we need to check with the original author to see why it was `YES` before? Are there any side effects?

**Review by RickWinter — APPROVED** [review]
This looks fine assuming Enums are still showing up in documentation correctly

**Comment by vhvb1989** [comment]
> This looks fine assuming Enums are still showing up in documentation correctly

It also fixes enums.  Sharing the before and after below:

## Before the change 

### Class

image

### Enum

image


## After

### Class

image


## Enum

image
